CHANGES/8163.bugfix.rst

```diff
@@ -0,0 +1 @@
+Avoid creating a task to do DNS resolution if there is no throttle
```

Comment:
Check out the recently improved guidelines: https://docs.aiohttp.org/en/latest/contributing.html#changelog-update / https://github.com/aio-libs/aiohttp/tree/master/CHANGES#readme.

Perhaps, something like this would be easier comprehend for the end-user audience:
```suggestion
Improved the not-throttled DNS resolution performance
-- by :user:`bdraco`.

This is achieverd by avoiding an :mod:`asyncio` task creation
in this case.
```

Reply:
Yep, it's hard for everyone which is why I'm failing to come up with better guidelines :)

I just attempted rephrasing what you posted in the PR, that's all my context now.

Reply:
"Performance: Avoided creating tasks unnecessarily during DNS resolution."

would cover it for me. I don't even know what throttled means in this context.

Reply:
I adjusted it again to remove the world throttled

Reply:
That's too low-level. Throttling refers to limiting the number of simultaneous connections to resolve the same hostname, it seems.

Reply:
> Throttling refers to limiting the number of simultaneous connections to resolve the same hostname, it seems.

Right, to me that's just sensible caching (e.g. If you just added `@alru_cache` to the function it would also do that).

Reply:
Brilliant! That's much better!